source/manuals/programming-languages/nodejs/style-guide.html.md.erb

```diff
@@ -0,0 +1,119 @@
+---
+title: GDS Node.js Style Guide
+---
+
+# <%= current_page.data.title %>
+
+## General style
+
+As with all JavaScript at GDS, for Node.js applications we use the [standard] style.
+
+## Use strict mode
+
+[Strict mode] should be enabled for Node.js modules, as in the browser this prevents polluting the global scope (in
+Node.js this would pollute the `global` object, rather than the `window` object).
+
+## ES2015+
+
+Because with Node.js you control the version of the interpreter you're code is running on,
+you have access to many the latest features of JavaScript, such as:
+
+- [let] and [const]
+- [Arrow functions]
+- [Array.prototype] methods like `Array.map()`, `Array.find()` and `Array.values()`
+- The [spread operator]
+
+However use caution when using cutting edge JavaScript features and be sure to read the [documentation].
+
+A full list of what features are supported in each version of Node.js can be found at [Node Green].
+
+
+### let and const
+
+For Node.js code, be explicit about variable mutability using [const] and [let]:
+
+```
+const foo // GOOD: explicitly immutable variable declaration
+let bar // GOOD: expicitly mutable variable declaration
+var baz // BAD: not explicit about mutability of variable
```

Comment:
Are there cases where the use of var would be applicable? For example you could use a var keyword to convey to a developer that this variable will be used across a whole function, rather than scoping it to a particular block (if / else etc).

Reply:
I would still say it's still better to use let to convey this usage, except declared at the top of the function. See below example:
```
function doSomething(someBool) {
	let variableUsedAcrossFunction
	if (someBool) {
		variableUsedAcrossFunction = 'world'
	} else {
		variableUsedAcrossFunction = 'planet earth'
	}
	return `hello ${variableUsedAcrossFunction}`
}
```

Reply:
Also, it is explicit about mutability no? vars can be changed, consts can't. I think the point is more about scope

Reply:
Yeah the point is more around scope. It is possible to use a mixture of let and vars in a way that you convey more meaning rather that explicitly using only one or the other.

For example if setup correctly:
- I see a `let` keyword I know that it's usage is restricted to the block it is set (by design).
- I see a `var` keyword I now know that this is being used across the whole function. 

Simple example:
````
// use both let and var together as it means different things
function foo(x,y) {
    
    try {
        // because we have used let, z is bound to this block
        let z = bar( x * 2 );
    } catch(e){
        //..
    }
    
    // we will use this across the function
    var z = x * 4;
    
    if(x > y){
        // scoped to this block
        let tmp = x;
        x = y;
        y = tmp;
    }
    
    // i scoped to this loop
    for(let i = 0;i< 10; i++){
        //...
    }
    
}
````

I know this method is favoured by Kyle Simpson from the "You don't know JS" book series so I just raised as something that may be worth considering.

Reply:
Being not Node.js expert, coming from other languages (Java) I would like to comment as well:
For me it's more clearer to use `let` on the top of the function, instead of `var` and to be aware that `var` actually is visible in the whole function scope.
I can see advantages of using @Nooshu 's suggestion, but still remembering the use cases for `var` instead of declaring the variables on the top of the function using `let` is more "risky" when the person doesn't know the use cases of `var`.
This is only opinion from non native Node.js developer :)
Thanks